diff -rupN tuya-iot-core-sdk/src/tuyalink_core.c tuya-iot-core-sdk-user-var/src/tuyalink_core.c
--- tuya-iot-core-sdk/src/tuyalink_core.c	2024-03-05 08:37:51.577433059 +0200
+++ tuya-iot-core-sdk-user-var/src/tuyalink_core.c	2024-03-14 11:49:54.280889983 +0200
@@ -473,6 +473,7 @@ int tuya_mqtt_init(tuya_mqtt_context_t*
 
 	/* configuration */
 	context->config = *config;
+	context->user_data = config->user_data;
 
 	/* Device token signature */
 	rt = tuya_mqtt_auth_signature_calculate(config->device_id, config->device_secret,
@@ -1068,4 +1069,4 @@ int tuyalink_tag_delete(tuya_mqtt_contex
 		.ack = false
 	};
 	return tuyalink_message_send(context, &message);
-}
\ No newline at end of file
+}
